### PR TITLE
Free unused memory in Data.from

### PR DESCRIPTION
**Pull request by yHSJ**
At JPG Store, we have been dealing with memory leaks in our code that uses either Lucid or the CML. After some research, we discovered that the CML does a lot of `clone`s to avoid issues with freed memory across the WASM boundary ( https://github.com/dcSpark/cardano-multiplatform-lib/issues/142#issuecomment-1281810732). 

Any object that is returned from WASM to JavaScript seems to never get garbage collected. Thus, **all** objects returned from wasm to the js runtime must be freed, including intermediary values when chaining function calls together.

This PR specifically only fixes  the memory usage in the `Data.from` function to resolve the following issue: https://github.com/spacebudz/lucid/issues/222

I will continue to fix the memory management over time, as I have free time. However, I would argue it is best to merge these smaller pieces whenever they are done, as they will reduce the memory footprint of several applications relying on Lucid.

### CODE REVIEW

File: src/utils/freeable.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export interface Freeable {
+  free(): void;
+}
+
+export type FreeableBucket = Array<Freeable | undefined>;
```
Comment (joacohoyos) [inline src/utils/freeable.ts:5]: Make this so it's also nullable. I've seen cases on jpg where I had to add null also

File: src/utils/freeable.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+export abstract class Freeables {
+  static free(...bucket: (Freeable | undefined)[]) {
```
Comment (joacohoyos) [inline src/utils/freeable.ts:8]: same here add null

**Review by joacohoyos — CHANGES_REQUESTED** [review]
Small comment. We might also want to add a comment to the freeable class cause other people might not have the context we have due to our internal discussions

**Comment by yHSJ** [comment]
Closing in favor of https://github.com/spacebudz/lucid/pull/234 which includes managing WASM memory everywhere it's used.

**Pull request closed without merging**